pkg/prometheus/agent/statefulset_test.go

```diff
@@ -52,75 +52,29 @@ func TestListenTLS(t *testing.T) {
 }
 
 func TestWALCompression(t *testing.T) {
-	tests := []struct {
-		version       string
-		enabled       *bool
-		expectedArg   string
-		shouldContain bool
-	}{
-		// Nil should not have either flag.
-		{"v2.30.0", ptr.To(false), "--storage.agent.wal-compression", false},
-		{"v2.32.0", nil, "--storage.agent.wal-compression", false},
-		{"v2.32.0", ptr.To(false), "--no-storage.agent.wal-compression", true},
-		{"v2.32.0", ptr.To(true), "--storage.agent.wal-compression", true},
-	}
+	testcases := createTestCasesForTestWALCompression()
 
-	for _, test := range tests {
-		sset, err := makeStatefulSetFromPrometheus(monitoringv1alpha1.PrometheusAgent{
-			Spec: monitoringv1alpha1.PrometheusAgentSpec{
-				CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
-					Version:        test.version,
-					WALCompression: test.enabled,
-				},
-			},
-		})
+	for _, test := range testcases {
+		sset, err := makeStatefulSetFromPrometheus(
+			makePrometheusAgentForTestWALCompression(
+				test.version,
+				test.enabled))
```

Comment:
same comment here... we're writing 10 lines for `makePrometheusAgentForTestWalCompression`, just to hide 2 lines here. Is it really worth it?